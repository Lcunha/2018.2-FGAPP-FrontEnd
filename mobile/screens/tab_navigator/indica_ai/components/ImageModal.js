import React from 'react';
import {
  View,
  Text,
  Modal,
  TextInput,
  StyleSheet,
  TouchableOpacity
} from 'react-native';
import Icon from 'react-native-vector-icons/SimpleLineIcons';

export default class ImageModal extends React.Component {

  render() {
    return (
      <Modal animationType="fade" transparent={true} visible={this.props.visible} onRequestClose={() => null}>
        <View style={styles.modalContainer}>

          <View style={styles.boxContainer}>
            <Text style={styles.boxTitle}>Enviar foto ?</Text>
            <View style={styles.buttonContainer}>

              <TouchableOpacity style={[styles.button, styles.sendButton]} onPress={this.props.onSendImage}>
                <Text style={styles.buttonText}>Enviar</Text>
              </TouchableOpacity>

              <TouchableOpacity>
                <Text style={styles.buttonText}>Cancelar</Text>
              </TouchableOpacity>

            </View>
          </View>
        </View>
      </Modal>
    );
  }
}

const styles = StyleSheet.create({

  modalContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: 'rgba(0,0,0,0.5)'
  },
  boxContainer: {
    padding: 20,
    backgroundColor: '#FFF',
    borderRadius: 10,
    alignItems: 'center',
    width: 280,
  },
  boxTitle: {
    fontWeight: 'bold',
    fontSize: 16,
    marginTop: 8
  },
  buttonContainer: {
    marginTop: 10,
    height: 40,
    flexDirection: 'row',
  },
  button: {
    flex: 1,
    alignItems: 'flex-start',
    justifyContent: 'flex-start',
    borderRadius: 3,
  },
  sendButton: {
    backgroundColor: 'green',
  },
  cancelButton: {
    backgroundColor: 'red',
    marginLeft: 10,

  },
  buttonText: {
    fontWeight: 'bold',
    color: '#FFF',
    fontSize: 12,
  },

});
