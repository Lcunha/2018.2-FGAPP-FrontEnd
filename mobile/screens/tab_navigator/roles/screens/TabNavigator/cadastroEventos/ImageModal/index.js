import React, { Component } from "react";
import { Card, CardItem, Body, Icon, Button, Input } from "native-base";
import {
	ScrollView,
	Text,
	StyleSheet,
	TouchableOpacity,
	View,
	TouchableHighlight,
	Image
} from "react-native";
import Modal from "react-native-modal";

export default class ImageModal extends Component {
	render() {
		return (
			<Modal
				isVisible={this.props.visible}
				onBackButtonPress={this.props.closeModal}
				onBackdropPress={this.props.closeModal}
				animationIn="fadeInUp"
				animationOut="fadeOutDown"
			>
				<Card styles={styles.modalCard}>
					<View>
						<Image
							source={{ uri: this.props.photoSrc }}
							style={{
								height: 300,
								width: "100%"
							}}
						/>
            <Button
              block
              dark  
              bordered
              onPress={()=>this.props.editBtn()}
            >
              <Icon type='Feather' name='edit'/>
              <Text>Enviar outra imagem</Text>
            </Button>
					</View>
				</Card>
			</Modal>
		);
	}
}

const styles = StyleSheet.create({
	modalCard: {
		borderRadius: 10,
		width: "100%",
		alignSelf: "center",
		alignItems: "center"
  },
});
