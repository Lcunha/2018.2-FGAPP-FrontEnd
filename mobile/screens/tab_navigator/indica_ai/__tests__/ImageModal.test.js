import React from 'react';
import Adapter from 'enzyme-adapter-react-16';
import {shallow} from 'enzyme';
import Enzyme from 'enzyme';
import renderer from 'react-test-renderer';

import ImageModal from '../components/ImageModal';

Enzyme.configure({adapter: new Adapter()});

  it('renders correctly', () => {
    const tree = renderer.create(<ImageModal />).toJSON();
    expect(tree).toMatchSnapshot();
});
