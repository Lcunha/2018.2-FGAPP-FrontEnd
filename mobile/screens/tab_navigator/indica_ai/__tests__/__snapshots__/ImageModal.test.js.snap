// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0,0,0,0.5)",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#FFF",
          "borderRadius": 10,
          "padding": 20,
          "width": 240,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 16,
            "fontWeight": "bold",
            "marginTop": 8,
          }
        }
      >
        Enviar foto ?
      </Text>
      <View
        style={
          Object {
            "flexDirection": "row",
            "height": 40,
            "marginTop": 10,
            "width": 130,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "green",
              "borderRadius": 5,
              "justifyContent": "center",
              "opacity": 1,
              "width": 65,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignItems": "center",
                "color": "#FFF",
                "fontSize": 15,
                "fontWeight": "bold",
              }
            }
          >
            Enviar
          </Text>
        </View>
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "red",
              "borderRadius": 5,
              "justifyContent": "center",
              "marginLeft": 10,
              "opacity": 1,
              "width": 65,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Text
            accessible={true}
            allowFontScaling={true}
            ellipsizeMode="tail"
            style={
              Object {
                "alignItems": "center",
                "color": "#FFF",
                "fontSize": 15,
                "fontWeight": "bold",
              }
            }
          >
            Cancelar
          </Text>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
